### PR TITLE
fix: fix loadSider false render structure issue

### LINKED ISSUE #519: Fix inconsistent sidebar nesting
The sidebar nesting logic is inconsistent which can create challenges for plugin and theme authors looking to target specific navigation levels.

Here is the docsify HTML output from a three-level sidebar navigation:

```html
<div class="sidebar-nav">
  <ul>
    <li class="active">
      <a href="#/page">Page</a>
      <ul class="app-sub-sidebar">
        <li>
          <a class="section-link" href="#/page?id=section">Section</a>
        </li>
        <li>
          <ul class="children">
            <li>
              <a class="section-link" href="#/page?id=link">Link</a>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ul>
</div>
```

Take notice of how the top-level item and its children are rendered:

```html
<li class="active">
  <a href="#/page">Page</a>
  <ul class="app-sub-sidebar">
    ...
  </ul>
</li>
```

The child items are rendered as a `ul` element which is a sibling of the `a` element. This makes it easy to target the `ul` element via CSS (`a + ul`) or JavaScript (`aNode.nextSibling`).

Now take notice of how the second level links are rendered:

```html
<li>
  <a class="section-link" href="#/page?id=section">Section</a>
</li>
<li>
  <ul class="children">
    ...
  </ul>
</li>
```

Here the `a` and `ul` elements are contained in separate `li` items. This makes it impossible to target the `ul` from the `a` element via CSS, and more expensive via JavaScript (`aNode.parentNode.nextSibling.querySelector('ul.children')`). This also breaks the default list styling, as the sibling `li` elements render as follows:

```
• Page
  • Section
  • 
    • Child
    • Child
```

What I would hope to see is the same nesting logic applied to all levels of the sidebar navigation: links that have children are rendered as an `a` tag immediately followed by a `ul` tag containing the children:

```html
<li>
  <a class="section-link" href="#/page?id=section">Section</a>
  <ul class="children">
    ...
  </ul>
</li>
```

Using this nesting logic would make targeting via CSS and JS much easier, and allow the default list styling to work for an infinite number of levels:

```
• Level 1
  • Level 2
    • Level 3
      • Level 4
        ...
```

Thanks!

### PR DESCRIPTION
**Pull request by Koooooo-7**
## Summary

Enhance fix #519 .
#### Main fix

When the `loadSidebar=false`, it uses the `toc` to render auto-generated sidebar.
But it puts the `children nodes` contents out of the parent node `<li>` block, which is mismatch the behavior with `loadSidebar=true`.

#### Minor changes

Refine the sidebar render part to always trigger by `_renderMain` callback instead of mess it in `#renderMain`.
Now, the renderSidebar has the consistency that always being a callback after `#renderMain` no matter it load on `loadSidebar=true/false`.
  

## What kind of change does this PR introduce?

Bugfix
## For any code change,

- [ ] Related documentation has been updated, if needed
- [x] Related tests have been added or updated, if needed

## Does this PR introduce a breaking change?

Yes
No

## Tested in the following browsers:

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

**Comment by jhildenbiddle** [comment]
@Koooooo-7 --

Tested this PR today. The nesting appears to be fixed (🥳), however a bug has been introduced where it appears that an attempt is made to load a sidebar file named `null.md` even when `loadSidebar: true`. You can see this by looking at the network tab of your browser's dev tools. The failing e2e test also shows the error message appearing in the console:

```
       "beforeEach",
        "afterEach-async",
        "afterEach",
        "doneEach",
        "ready",
    +   "Failed to load resource: the server responded with a status of 404 (Not Found)",
```

**Comment by Koooooo-7** [comment]
> however a bug has been introduced where it appears that an attempt is made to load a sidebar file named `null.md` even when `loadSidebar: true`. 

@jhildenbiddle  ...nice catch. I got a silly mistake to re-load the sidebar twice.
Fixed.

**Review by jhildenbiddle — CHANGES_REQUESTED** [review]
A few issues I've found:

1. When loading sidebars from a file (`loadSidebar: true`), there are now multiple `.app-sub-sidebar` elements when there should only be one (the `<ul>` that contains the links generated from page headings). I'm guessing this is because the rendering loop is adding the `app-sub-sidebar` class to each `<ul>`element (top-level and each nested list).

   

2. When generating sidebars without a file (`loadSidebar: false`), there is no `app-sub-sidebar` element. As mentioned above, the `<ul>` element that links generated from page headings should always have the `app-sub-sidebar` class name applied.

   

3. Separate but related, it appears #2467 introduced a new behavior that results in the top-level `<h1>` tag appearing as a nested link. This isn't necessarily *wrong*, but it is a new and unexpected behavior. I mention it here only because it seems related to the work being done here.

   **Docsify v4**
   
   

   **Docsify v5 style overhaul** (#2469 Preview)
   
   
   
   **This PR**

File: test/integration/sidebar.test.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
+import docsifyInit from '../helpers/docsify-init.js';
+import { expect } from '@playwright/test';
```
[the remark this answers is not in the document]
Comment (Koooooo-7) [reply test/integration/sidebar.test.js:2]: I use the wrong function import. fix.

**Comment by Koooooo-7** [comment]
@jhildenbiddle  ---
Fix the  issue 1 and 3.
The Issue 2, once  we have a final decision I will update asap.

**Review by jhildenbiddle — APPROVED** [review]
LGTM! Thank you, @Koooooo-7!

**Review by paulhibbitts — APPROVED** [review]
Thanks so much @Koooooo-7 , looks good with my testing with Docsify-This as well.

**Pull request merged**